packages/aws-cdk/lib/toolkit/cli-io-host.ts

```diff
@@ -335,12 +338,57 @@ export class CliIoHost implements IIoHost {
    * default response from the input message will be used.
    */
   public async requestResponse<T, U>(msg: IoRequest<T, U>): Promise<U> {
+    // First call out to a registered instance if we have one
     if (this._internalIoHost) {
       return this._internalIoHost.requestResponse(msg);
     }
 
-    await this.notify(msg);
-    return msg.defaultResponse;
+    // If the request cannot be prompted for by the CliIoHost, we just accept the default
+    if (!isPromptableRequest(msg)) {
+      await this.notify(msg);
+      return msg.defaultResponse;
+    }
+
+    // prepare prompt data
+    // @todo this format is not defined anywhere, probably should be
+    const data: {
+      motivation?: string;
+      concurrency?: number;
+    } = msg.data ?? {};
+
+    const motivation = data.motivation ?? 'User input is needed';
+    const concurrency = data.concurrency ?? 0;
+
+    const prompt = extractPrompt(msg);
+    const preview = prompt.preview ? ` (${prompt.preview})` : '';
```

Comment:
nit: i don't like the word preview, is that standard somewhere? to me, something like `options` would make more sense.